Frameworks/Foundation/_NSInvocation.arm.asm

```diff
@@ -84,4 +84,196 @@
+
+    MOV r2, r1 ; sel
+    MOV r1, r0 ; self
+    MOV r0, #0 ; stret
+
+    ; Skip to the body of _NSInvocation_ForwardingBridgeNoStret
+    B %F0
+
+    ; Second entrypoint to this function; called when there is a struct return pointer.
+    ; r0 = struct return pointer
+    ; r1 = self
+    ; r2 = _cmd
+|_NSInvocation_ForwardingBridge|
+    ;;; This code must match the layout of _NSInvocation_ForwardingBridge.prolog.xdata below.
+    PUSH {r0-r3}
+    VSTMDB sp!, {d0-d7}
+    MOV r3, sp
+    PUSH {r4, fp, lr}
+    SUB sp, sp, #12
+|_NSInvocation_ForwardingBridge.prolog.end|
+
+0
+|_NSInvocation_ForwardingBridge.partial1.start|
+    ADD fp, sp, #16
+    ADD r4, sp, #4
+    STR r4, [sp]
+    BLX _NSInvocation_ForwardFrame ; (stret, self, sel, frame, bridgeInfo)
+
+    LDR r3, [r4, #4] ; return type
+    LDR r2, [r4, #0] ; return pointer
+
+|_NSInvocation_ForwardingBridge.partial1.end|
+|_NSInvocation_ForwardingBridge.epilog1.start|
+    ;;; This code must match the layout of _NSInvocation_ForwardingBridge.epilog1.xdata below.
+    ADD sp, sp, #12 ; Consume r4 + bridgeInfo
+    POP {r4, fp, lr}
+    ; Consume the stored GP and VFP registers
+    ADD sp, sp, #0x50 ; 0x40 for VFP, 0x10 for GPR
+|_NSInvocation_ForwardingBridge.epilog1.end|
+
+    ADR ip, %F101
+    TBB [ip, r3]
+
+100
+111 ; VFP_F
+    VLDR s0, [r2]
+    BX lr
+112 ; VFP_D
+    VLDR d0, [r2]
+    BX lr
+113 ; VFP_HOMOGENOUS
+    VLDM r2, {d0-d3}
+    BX lr
+
+115 ; INT64
+    LDR r1, [r2, #4]
+    ;; fall through
+114 ; INT32
+    LDR r0, [r2]
+    ;; fall through
+
+116 ; STRUCT
+110 ; NONE
+    BX lr
+
+101 ; Must be kept in the order from _NSInvocation.arm.mm
+    DCB (%B110-%B100)/2 ; RETURN_TYPE_NONE
+    DCB (%B111-%B100)/2 ; RETURN_TYPE_VFP_S
+    DCB (%B112-%B100)/2 ; RETURN_TYPE_VFP_D
+    DCB (%B113-%B100)/2 ; RETURN_TYPE_VFP_HOMOGENOUS
+    DCB (%B114-%B100)/2 ; RETURN_TYPE_INT
+    DCB (%B115-%B100)/2 ; RETURN_TYPE_INT64
+    DCB (%B116-%B100)/2 ; RETURN_TYPE_STRUCT
+
+    ;;; End of functions _NSInvocation_ForwardingBridge and _NSInvocation_ForwardingBridgeNoStret
+    ;;; Beginning of unwind/exception data.
+
+    ; .pdata and .xdata are laid out as described in ARM Exception Handling (https://msdn.microsoft.com/en-us/library/dn743843.aspx)
+    ; Please refer to this document for a comprehensive look at ARM exception unwinding.
+
+    AREA |.pdata|,ALIGN=4,DATA,READONLY
+    ALIGN 4
+    DCD |_NSInvocation_ForwardingBridgeNoStret|
+    RELOC 2
+    DCD |_NSInvocation_ForwardingBridgeNoStret.prolog.xdata|
+    RELOC 2
+
+    DCD |_NSInvocation_ForwardingBridge|
+    RELOC 2
+    DCD |_NSInvocation_ForwardingBridge.prolog.xdata|
+    RELOC 2
+
+    DCD |_NSInvocation_ForwardingBridge.partial1.start|
+    RELOC 2
+    DCD |_NSInvocation_ForwardingBridge.partial1.xdata|
+    RELOC 2
+
+    DCD |_NSInvocation_ForwardingBridge.epilog1.start|
+    RELOC 2
+    DCD |_NSInvocation_ForwardingBridge.epilog1.xdata|
+    RELOC 2
+
+    LTORG
```

Comment:
This was from the macros in kxarm, which we're now using directly.

---
In reply to: [77705648](https://github.com/Microsoft/WinObjC/pull/951#discussion_r77705648) [](ancestors = 77705648)